server/tests/common/fixture.rs

```diff
@@ -0,0 +1,342 @@
+use anyhow::{anyhow, bail, Context, Result};
+use assert_cmd::prelude::*;
+use graphql_client::GraphQLQuery;
+use reqwest::blocking::{Client, ClientBuilder};
+use std::collections::{HashMap, HashSet};
+use std::process::{Child, Command};
+use std::{env::var, fs::canonicalize, thread, time::Duration};
+
+const DB_KEY: &str = "LLDAP_DATABASE_URL";
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/add_user_to_group.graphql",
+    response_derives = "Debug",
+    variables_derives = "Debug,Clone",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+struct AddUserToGroup;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/create_user.graphql",
+    response_derives = "Debug",
+    variables_derives = "Debug,Clone",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+struct CreateUser;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/create_group.graphql",
+    response_derives = "Debug",
+    variables_derives = "Debug,Clone",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+struct CreateGroup;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/list_users.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+pub struct ListUsers;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/list_groups.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+struct ListGroups;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/delete_group.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+struct DeleteGroupQuery;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "../schema.graphql",
+    query_path = "tests/queries/delete_user.graphql",
+    response_derives = "Debug",
+    custom_scalars_module = "crate::infra::graphql"
+)]
+struct DeleteUserQuery;
+
+#[derive(Clone)]
+pub struct User {
+    pub username: String,
+    pub groups: Vec<String>,
+}
+
+impl User {
+    pub fn new(username: &str, groups: Vec<&str>) -> Self {
+        let username = username.to_string();
+        let groups = groups.iter().map(|username| username.to_string()).collect();
+        Self { username, groups }
+    }
+}
+
+pub struct LLDAPFixture {
+    token: String,
+    client: Client,
+    child: Child,
+    users: HashSet<String>,
+    groups: HashMap<String, i64>,
+}
+
+impl LLDAPFixture {
+    pub fn new() -> Self {
+        let mut cmd = Command::cargo_bin(env!("CARGO_PKG_NAME")).expect("cargo bin found");
```

Comment:
Note on expect: the message appears in case of failures, so you want "cargo bin not found"

Reply:
Will change - it was one of the first expects i wrote before realized it's inverted. It's a bit of a habit i need to break